src/pharmpy/model/external/nonmem/advan.py

```diff
@@ -590,36 +595,92 @@ def _advan12_trans(trans: str):
 
 
 def dosing(di: DataInfo, dataset, dose_comp: int):
-    if 'CMT' not in di.names or di['CMT'].drop or dataset is None:
-        return ({'comp_number': dose_comp, 'dose': _dosing(di, dataset, dose_comp)},)
-    else:
-        # CMT column present
-        cmt = dataset['CMT']
-        if len(cmt.unique()) < 1:
-            # Single compartment dose
-            # Overwrite dose_comp
+    # Only check doses
+    if dataset is not None:
+        dataset = dataset[dataset['AMT'] != 0]
+
+    cmt_loop = False
+    admid_loop = False
+    return_dose = False
+    if dataset is None:
+        return_dose = True
+    elif 'ADMID' in di.names:
```

Comment:
The name ADMID shouldn't be assumed. Look for the 'admid' type using the type indexer. ('CMT' is safe to assume since it is a reserved name)